activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1185,7 +1185,7 @@ def where_clause_factory
       end
 
       Relation::SINGLE_VALUE_METHODS.each do |value|
-        DEFAULT_VALUES[value] = nil if DEFAULT_VALUES[value].nil?
+        DEFAULT_VALUES[value] = nil
```

Comment:
I would have done `DEFAULT_VALUES[value] ||= nil` but when because the `:readonly` key is false, it wouldn't work. 

We could do `Relation::SINGLE_VALUE_METHODS.except(:readonly)` if it makes it clearer